### ISSUE TITLE
Can't sync chain because of DNS error `cannot unmarshal DNS message`

### ISSUE THREAD

**Issue by ziogaschr** [etclabscore/core-geth#262]
#### System information

```
CoreGeth
Version: 1.11.20-unstable
Architecture: amd64
Protocol Versions: [65 64 63]
Go Version: go1.15.5
Operating System: darwin
GOPATH=/Users/ziogaschr/Sites/go
GOROOT=/usr/local/Cellar/go/1.15.5/libexec
```

#### Expected behaviour

Chain will start syncing.

#### Actual behaviour

Chain fails to sync, or better it syncs slowly, losing peers very often.

#### Steps to reproduce the behaviour

```
geth --verbosity 5 --classic
```

#### Backtrace

Logs to check:

```
TRACE[12-22|18:13:03.406] DNS discovery lookup                     name=6AHZVTV7SHENRJ5VHJRKZ24QQE.all.classic.blockd.info err="lookup 6AHZVTV7SHENRJ5VHJRKZ24QQE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
DEBUG[12-22|18:13:03.406] Error in DNS random node sync            tree=all.classic.blockd.info err="lookup 6AHZVTV7SHENRJ5VHJRKZ24QQE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
```

Full backtrace:

```
INFO [12-22|18:13:02.247] Starting Geth on Ethereum Classic... 
DEBUG[12-22|18:13:02.248] Sanitizing Go's GC trigger               percent=50
INFO [12-22|18:13:02.250] Maximum peer count                       ETH=50 LES=0 total=50
DEBUG[12-22|18:13:02.250] FS scan times                            list="121.917µs" set="2.865µs" diff="2.355µs"
TRACE[12-22|18:13:02.251] Started watching keystore folder         path=/Volumes/Work/chain-datadirs/tmp.classic.fast/keystore
INFO [12-22|18:13:02.277] Set global gas cap                       cap=25000000
INFO [12-22|18:13:02.287] Allocated trie memory caches             clean=511.00MiB dirty=512.00MiB
INFO [12-22|18:13:02.287] Allocated cache and file handles         database=/Volumes/Work/chain-datadirs/tmp.classic.fast/geth/chaindata cache=1024.00MiB handles=5120
DEBUG[12-22|18:13:02.304] Chain freezer table opened               database=/Volumes/Work/chain-datadirs/tmp.classic.fast/geth/chaindata/ancient table=headers items=0 size=0.00B
DEBUG[12-22|18:13:02.307] Chain freezer table opened               database=/Volumes/Work/chain-datadirs/tmp.classic.fast/geth/chaindata/ancient table=hashes  items=0 size=0.00B
DEBUG[12-22|18:13:02.310] Chain freezer table opened               database=/Volumes/Work/chain-datadirs/tmp.classic.fast/geth/chaindata/ancient table=bodies  items=0 size=0.00B
DEBUG[12-22|18:13:02.313] Chain freezer table opened               database=/Volumes/Work/chain-datadirs/tmp.classic.fast/geth/chaindata/ancient table=receipts items=0 size=0.00B
DEBUG[12-22|18:13:02.317] Chain freezer table opened               database=/Volumes/Work/chain-datadirs/tmp.classic.fast/geth/chaindata/ancient table=diffs    items=0 size=0.00B
INFO [12-22|18:13:02.317] Opened ancient database                  database=/Volumes/Work/chain-datadirs/tmp.classic.fast/geth/chaindata/ancient
INFO [12-22|18:13:02.317] Writing custom genesis block 
DEBUG[12-22|18:13:02.317] Current full block hash unavailable 
INFO [12-22|18:13:02.500] Persisted trie from memory database      nodes=12356 size=1.78MiB time=45.863173ms gcnodes=0 gcsize=0.00B gctime=0s livenodes=1 livesize=0.00B
INFO [12-22|18:13:02.501] Wrote custom genesis block OK            config="NetworkID: 1, ChainID: 61 Engine: ethash ECBP1100: 11380000 EIP1014: 9573000 EIP1052: 9573000 EIP1108: 10500839 EIP1344: 10500839 EIP140: 8772000 EIP145: 9573000 EIP150: 2500000 EIP152: 10500839 EIP155: 3000000 EIP160: 3000000 EIP161abc: 8772000 EIP161d: 8772000 EIP170: 8772000 EIP1884: 10500839 EIP198: 8772000 EIP2028: 10500839 EIP211: 8772000 EIP212: 8772000 EIP213: 8772000 EIP214: 8772000 EIP2200: 10500839 EIP2: 1150000 EIP658: 8772000 EIP7: 1150000 EthashECIP1010Continue: 5000000 EthashECIP1010Pause: 3000000 EthashECIP1017: 5000000 EthashECIP1041: 5900000 EthashECIP1099: 11700000 EthashEIP100B: 8772000 EthashHomestead: 1150000 "
INFO [12-22|18:13:02.501] Initialised chain configuration          config="NetworkID: 1, ChainID: 61 Engine: ethash ECBP1100: 11380000 EIP1014: 9573000 EIP1052: 9573000 EIP1108: 10500839 EIP1344: 10500839 EIP140: 8772000 EIP145: 9573000 EIP150: 2500000 EIP152: 10500839 EIP155: 3000000 EIP160: 3000000 EIP161abc: 8772000 EIP161d: 8772000 EIP170: 8772000 EIP1884: 10500839 EIP198: 8772000 EIP2028: 10500839 EIP211: 8772000 EIP212: 8772000 EIP213: 8772000 EIP214: 8772000 EIP2200: 10500839 EIP2: 1150000 EIP658: 8772000 EIP7: 1150000 EthashECIP1010Continue: 5000000 EthashECIP1010Pause: 3000000 EthashECIP1017: 5000000 EthashECIP1041: 5900000 EthashECIP1099: 11700000 EthashEIP100B: 8772000 EthashHomestead: 1150000 "
INFO [12-22|18:13:02.501] Disk storage enabled for ethash caches   dir=/Volumes/Work/chain-datadirs/tmp.classic.fast/geth/etchash count=3
INFO [12-22|18:13:02.501] Disk storage enabled for ethash DAGs     dir=/Users/ziogaschr/Library/Etchash count=2
INFO [12-22|18:13:02.501] Initialising Ethereum protocol           versions="[65 64 63]" network=1 dbversion=<nil>
WARN [12-22|18:13:02.501] Upgrade blockchain database version      from=<nil> to=8
INFO [12-22|18:13:02.502] Loaded most recent local header          number=0 hash="d4e567…cb8fa3" td=17179869184 age=51y8mo2w
INFO [12-22|18:13:02.502] Loaded most recent local full block      number=0 hash="d4e567…cb8fa3" td=17179869184 age=51y8mo2w
INFO [12-22|18:13:02.502] Loaded most recent local fast block      number=0 hash="d4e567…cb8fa3" td=17179869184 age=51y8mo2w
DEBUG[12-22|18:13:02.502] Reinjecting stale transactions           count=0
INFO [12-22|18:13:02.502] Regenerated local transaction journal    transactions=0 accounts=0
INFO [12-22|18:13:02.519] Allocated fast sync bloom                size=1023.00MiB
DEBUG[12-22|18:13:02.520] Recalculated downloader QoS values       rtt=20s confidence=1.000 ttl=1m0s
INFO [12-22|18:13:02.521] Starting peer-to-peer node               instance=CoreGeth/v1.11.20-unstable/darwin-amd64/go1.15.5
DEBUG[12-22|18:13:02.532] UDP listener up                          addr=[::]:30303
DEBUG[12-22|18:13:02.532] TCP listener up                          addr=[::]:30303
TRACE[12-22|18:13:02.532] Found seed node in database              id=1969d3adf6797e93 addr=159.203.56.33:30303 age=446848h13m2.532718s
INFO [12-22|18:13:02.533] New local node record                    seq=1 id=b678b50e62c565e1 ip=127.0.0.1 udp=30303 tcp=30303
TRACE[12-22|18:13:02.533] Found seed node in database              id=720108c959168462 addr=176.9.51.216:30355  age=446848h13m2.533392s
TRACE[12-22|18:13:02.533] Found seed node in database              id=9536874a9a1b4b51 addr=176.9.51.216:30365  age=446848h13m2.533434s
TRACE[12-22|18:13:02.533] Found seed node in database              id=6b90fe1510fa42c6 addr=18.205.66.229:30303 age=446848h13m2.533464s
TRACE[12-22|18:13:02.538] Found seed node in database              id=3064061e8cb4d357 addr=162.243.55.45:30303 age=446848h13m2.539003s
TRACE[12-22|18:13:02.539] Found seed node in database              id=a43b89f22ae65701 addr=104.237.131.102:30303 age=446848h13m2.539063s
TRACE[12-22|18:13:02.539] Found seed node in database              id=3becf3f13b090bcc addr=18.130.93.66:30303    age=446848h13m2.539089s
TRACE[12-22|18:13:02.539] Found seed node in database              id=9359cec51ae10cdb addr=178.62.238.75:30303   age=446848h13m2.53911s
TRACE[12-22|18:13:02.539] Found seed node in database              id=b6965dca741f86f0 addr=188.166.163.187:30303 age=446848h13m2.53913s
TRACE[12-22|18:13:02.539] Found seed node in database              id=d916ecf8224301e4 addr=46.101.78.44:30303    age=446848h13m2.539166s
TRACE[12-22|18:13:02.539] Found seed node in database              id=f5fb164a5c76dbb5 addr=68.183.99.5:30303     age=446848h13m2.539215s
TRACE[12-22|18:13:02.539] Found seed node in database              id=d4731bcf672459cc addr=206.189.68.191:30303  age=446848h13m2.539238s
TRACE[12-22|18:13:02.539] Found seed node in database              id=22ec854117125934 addr=159.89.169.103:30303  age=446848h13m2.539259s
TRACE[12-22|18:13:02.539] Found seed node in database              id=04a037eabb45f18c addr=178.128.55.119:30303  age=446848h13m2.539275s
TRACE[12-22|18:13:02.539] Found seed node in database              id=1887e1b2b420425c addr=35.187.57.94:30303    age=446848h13m2.539288s
TRACE[12-22|18:13:02.539] Found seed node in database              id=0970decfe54cafed addr=54.148.165.1:30303    age=446848h13m2.539317s
TRACE[12-22|18:13:02.539] Found seed node in database              id=27f254a8da5a1168 addr=54.148.165.1:30304    age=446848h13m2.539351s
TRACE[12-22|18:13:02.539] Found seed node in database              id=2ebda3f64fa1612a addr=123.57.29.99:30303    age=446848h13m2.539372s
TRACE[12-22|18:13:02.539] Found seed node in database              id=a597d00dd290d704 addr=47.108.52.30:30303    age=446848h13m2.53939s
TRACE[12-22|18:13:02.539] Found seed node in database              id=bd1203ac725a8910 addr=47.240.106.205:30303  age=446848h13m2.539408s
TRACE[12-22|18:13:02.539] Found seed node in database              id=51b59fdee3fc3ac3 addr=101.133.229.66:30303  age=446848h13m2.53947s
TRACE[12-22|18:13:02.539] Found seed node in database              id=059fce3b4d826ceb addr=47.115.150.90:30303   age=446848h13m2.539495s
TRACE[12-22|18:13:02.539] Found seed node in database              id=9e7ec22592c457be addr=47.91.30.48:30303     age=446848h13m2.539514s
TRACE[12-22|18:13:02.539] Starting p2p dial                        id=3064061e8cb4d357 ip=162.243.55.45 flag=dyndial
TRACE[12-22|18:13:02.539] Found seed node in database              id=1969d3adf6797e93 addr=159.203.56.33:30303   age=446848h13m2.539808s
TRACE[12-22|18:13:02.539] Found seed node in database              id=720108c959168462 addr=176.9.51.216:30355    age=446848h13m2.539836s
TRACE[12-22|18:13:02.539] Found seed node in database              id=9536874a9a1b4b51 addr=176.9.51.216:30365    age=446848h13m2.539854s
INFO [12-22|18:13:02.539] Started P2P networking                   self=enode://ae1fa4b6526f791bf36c9905b812d2b03b7d14f1b6eb2db7364efd8b95d0b83de14610bca8f944d5294af058569b4020d9a3fbf34a1d521e28199c6409ed58de@127.0.0.1:30303
TRACE[12-22|18:13:02.539] Found seed node in database              id=6b90fe1510fa42c6 addr=18.205.66.229:30303   age=446848h13m2.539872s
TRACE[12-22|18:13:02.539] Found seed node in database              id=3064061e8cb4d357 addr=162.243.55.45:30303   age=446848h13m2.539897s
TRACE[12-22|18:13:02.539] Found seed node in database              id=a43b89f22ae65701 addr=104.237.131.102:30303 age=446848h13m2.539965s
TRACE[12-22|18:13:02.539] Found seed node in database              id=3becf3f13b090bcc addr=18.130.93.66:30303    age=446848h13m2.539985s
TRACE[12-22|18:13:02.539] Found seed node in database              id=9359cec51ae10cdb addr=178.62.238.75:30303   age=446848h13m2.540049s
TRACE[12-22|18:13:02.540] Found seed node in database              id=b6965dca741f86f0 addr=188.166.163.187:30303 age=446848h13m2.540077s
TRACE[12-22|18:13:02.540] Found seed node in database              id=d916ecf8224301e4 addr=46.101.78.44:30303    age=446848h13m2.5401s
TRACE[12-22|18:13:02.540] Found seed node in database              id=f5fb164a5c76dbb5 addr=68.183.99.5:30303     age=446848h13m2.540123s
TRACE[12-22|18:13:02.540] Found seed node in database              id=d4731bcf672459cc addr=206.189.68.191:30303  age=446848h13m2.540142s
TRACE[12-22|18:13:02.540] Found seed node in database              id=22ec854117125934 addr=159.89.169.103:30303  age=446848h13m2.54016s
DEBUG[12-22|18:13:02.540] IPC registered                           namespace=admin
TRACE[12-22|18:13:02.543] Found seed node in database              id=04a037eabb45f18c addr=178.128.55.119:30303  age=446848h13m2.543804s
TRACE[12-22|18:13:02.543] Found seed node in database              id=1887e1b2b420425c addr=35.187.57.94:30303    age=446848h13m2.543925s
TRACE[12-22|18:13:02.543] Found seed node in database              id=0970decfe54cafed addr=54.148.165.1:30303    age=446848h13m2.543951s
TRACE[12-22|18:13:02.543] Found seed node in database              id=27f254a8da5a1168 addr=54.148.165.1:30304    age=446848h13m2.543971s
DEBUG[12-22|18:13:02.543] IPC registered                           namespace=admin
TRACE[12-22|18:13:02.543] Found seed node in database              id=2ebda3f64fa1612a addr=123.57.29.99:30303    age=446848h13m2.54399s
TRACE[12-22|18:13:02.544] Found seed node in database              id=a597d00dd290d704 addr=47.108.52.30:30303    age=446848h13m2.54401s
TRACE[12-22|18:13:02.544] Found seed node in database              id=bd1203ac725a8910 addr=47.240.106.205:30303  age=446848h13m2.544028s
TRACE[12-22|18:13:02.544] Found seed node in database              id=51b59fdee3fc3ac3 addr=101.133.229.66:30303  age=446848h13m2.544046s
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=debug
TRACE[12-22|18:13:02.544] Found seed node in database              id=059fce3b4d826ceb addr=47.115.150.90:30303   age=446848h13m2.544066s
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=web3
TRACE[12-22|18:13:02.544] Found seed node in database              id=9e7ec22592c457be addr=47.91.30.48:30303     age=446848h13m2.544084s
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=eth
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=eth
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=eth
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=txpool
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=debug
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=debug
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=eth
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=personal
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=eth
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=ethash
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=eth
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=eth
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=eth
TRACE[12-22|18:13:02.544] >> PING/v4                               id=a43b89f22ae65701 addr=104.237.131.102:30303 err=nil
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=miner
TRACE[12-22|18:13:02.544] >> PING/v4                               id=bd1203ac725a8910 addr=47.240.106.205:30303  err=nil
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=eth
TRACE[12-22|18:13:02.544] >> PING/v4                               id=b6965dca741f86f0 addr=188.166.163.187:30303 err=nil
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=admin
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=debug
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=debug
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=trace
DEBUG[12-22|18:13:02.544] IPC registered                           namespace=net
INFO [12-22|18:13:02.545] IPC endpoint opened                      url=/Volumes/Work/chain-datadirs/tmp.classic.fast/geth.ipc
DEBUG[12-22|18:13:02.545] Allowed origin(s) for WS RPC interface [*] 
INFO [12-22|18:13:02.545] HTTP server started                      endpoint=127.0.0.1:8545 cors=http://remix.ethereum.org vhosts=localhost
INFO [12-22|18:13:02.546] WebSocket enabled                        url=ws://127.0.0.1:8546
DEBUG[12-22|18:13:02.591] Served rpc_modules                       reqid=1 t="76.247µs"
TRACE[12-22|18:13:02.591] Handled RPC response                     reqid=1 t="1.265µs"
DEBUG[12-22|18:13:02.597] Served admin_datadir                     reqid=2 t="41.865µs"
TRACE[12-22|18:13:02.597] Handled RPC response                     reqid=2 t="1.238µs"
DEBUG[12-22|18:13:02.598] Served personal_listAccounts             reqid=3 t="32.117µs"
TRACE[12-22|18:13:02.598] Handled RPC response                     reqid=3 t=884ns
INFO [12-22|18:13:02.619] Initialized fast sync bloom              items=12356 errorrate=0.000 elapsed=99.846ms
TRACE[12-22|18:13:02.651] Updating DNS discovery root              tree=all.classic.blockd.info err=nil
TRACE[12-22|18:13:02.720] DNS discovery lookup                     name=FDXN3SN67NA5DKA4J2GOK7BVQI.all.classic.blockd.info err=nil
DEBUG[12-22|18:13:02.787] FS scan times                            list="118.719µs" set="1.102µs" diff="1.237µs"
TRACE[12-22|18:13:02.863] << PONG/v4                               id=bd1203ac725a8910 addr=47.240.106.205:30303  err=nil
TRACE[12-22|18:13:02.863] << PING/v4                               id=bd1203ac725a8910 addr=47.240.106.205:30303  err=nil
TRACE[12-22|18:13:02.863] DNS discovery lookup                     name=LB7XC5HGUHM7OTXE2ADPBQYPRA.all.classic.blockd.info err=nil
TRACE[12-22|18:13:02.863] >> PONG/v4                               id=bd1203ac725a8910 addr=47.240.106.205:30303  err=nil
TRACE[12-22|18:13:02.891] DNS discovery lookup                     name=V7HUH7LIEEIE5B6OMCYKYXF2I4.all.classic.blockd.info err=nil
TRACE[12-22|18:13:03.142] DNS discovery lookup                     name=FUI3BIYK4C5XQVPG74XW6YZQI4.all.classic.blockd.info err="lookup FUI3BIYK4C5XQVPG74XW6YZQI4.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
DEBUG[12-22|18:13:03.142] Error in DNS random node sync            tree=all.classic.blockd.info err="lookup FUI3BIYK4C5XQVPG74XW6YZQI4.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
TRACE[12-22|18:13:03.239] DNS discovery lookup                     name=6AHZVTV7SHENRJ5VHJRKZ24QQE.all.classic.blockd.info err="lookup 6AHZVTV7SHENRJ5VHJRKZ24QQE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
DEBUG[12-22|18:13:03.239] Error in DNS random node sync            tree=all.classic.blockd.info err="lookup 6AHZVTV7SHENRJ5VHJRKZ24QQE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
TRACE[12-22|18:13:03.317] DNS discovery lookup                     name=45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info err="lookup 45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
DEBUG[12-22|18:13:03.317] Error in DNS random node sync            tree=all.classic.blockd.info err="lookup 45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
TRACE[12-22|18:13:03.364] >> FINDNODE/v4                           id=bd1203ac725a8910 addr=47.240.106.205:30303  err=nil
TRACE[12-22|18:13:03.406] DNS discovery lookup                     name=6AHZVTV7SHENRJ5VHJRKZ24QQE.all.classic.blockd.info err="lookup 6AHZVTV7SHENRJ5VHJRKZ24QQE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
DEBUG[12-22|18:13:03.406] Error in DNS random node sync            tree=all.classic.blockd.info err="lookup 6AHZVTV7SHENRJ5VHJRKZ24QQE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
TRACE[12-22|18:13:03.480] DNS discovery lookup                     name=ATQH2K4ASJTRDH2JJJQL5VUHNM.all.classic.blockd.info err="lookup ATQH2K4ASJTRDH2JJJQL5VUHNM.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
DEBUG[12-22|18:13:03.480] Error in DNS random node sync            tree=all.classic.blockd.info err="lookup ATQH2K4ASJTRDH2JJJQL5VUHNM.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
TRACE[12-22|18:13:03.506] DNS discovery lookup                     name=45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info err="lookup 45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
DEBUG[12-22|18:13:03.506] Error in DNS random node sync            tree=all.classic.blockd.info err="lookup 45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
TRACE[12-22|18:13:03.521] Updating DNS discovery root              tree=all.classic.blockd.info err=nil
TRACE[12-22|18:13:03.541] DNS discovery lookup                     name=FUI3BIYK4C5XQVPG74XW6YZQI4.all.classic.blockd.info err="lookup FUI3BIYK4C5XQVPG74XW6YZQI4.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
DEBUG[12-22|18:13:03.541] Error in DNS random node sync            tree=all.classic.blockd.info err="lookup FUI3BIYK4C5XQVPG74XW6YZQI4.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
TRACE[12-22|18:13:03.546] >> FINDNODE/v4                           id=a43b89f22ae65701 addr=104.237.131.102:30303 err=nil
TRACE[12-22|18:13:03.547] >> FINDNODE/v4                           id=b6965dca741f86f0 addr=188.166.163.187:30303 err=nil
TRACE[12-22|18:13:03.563] DNS discovery lookup                     name=45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info err="lookup 45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
DEBUG[12-22|18:13:03.563] Error in DNS random node sync            tree=all.classic.blockd.info err="lookup 45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
````

**Comment by ziogaschr** [comment]
In order to examine the error better without all the node clutter, we can reproduce it using:

```
devp2p --verbosity 5 dns sync --timeout 10s enrtree://AJE62Q4DUX4QMMXEHCSSCSC65TDHZYSMONSD64P3WULVLSF6MRQ3K@all.classic.blockd.info
```

Which gives the following failing log:

```
TRACE[12-22|18:11:12.728] Updating DNS discovery root              tree=all.classic.blockd.info err=nil
TRACE[12-22|18:11:12.736] DNS discovery lookup                     name=FDXN3SN67NA5DKA4J2GOK7BVQI.all.classic.blockd.info err=nil
TRACE[12-22|18:11:12.772] DNS discovery lookup                     name=LB7XC5HGUHM7OTXE2ADPBQYPRA.all.classic.blockd.info err=nil
TRACE[12-22|18:11:12.787] DNS discovery lookup                     name=45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info err="lookup 45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message"
lookup 45WAU3H6PGBALIAO4OCBMCOHEE.all.classic.blockd.info on 192.168.1.18:53: cannot unmarshal DNS message
```

**Comment by ziogaschr** [comment]
By researching for this error we found out references that the:
* problem might reside in the go `net` package implementation
* DNS server should not send messages greater than 512 bytes over UDP [RFC 1035](https://tools.ietf.org/html/rfc1035).

In my home network I am using a **Mikrotik router** with custom configuration which might have issues. For this reason, I tried connecting to my phone **hotspot (4G) and check if it reproduces**. And magically it **works out nicely**.

Back to Mikrotik and with some further investigation, I figured out that under DNS settings I had the `Max UDP packet size = 4096`. By setting this value to `512` it worked like a charm.

But hey, how this works, the message is still >512. Haven't researched it more than that as much time has been spent, but from what I read when this setting is set to lower value than what comes in, it converts UDP to TCP. So it will work as UDP for <=512 bytes but will use TCP above it.

I hope the solution will help someone. Cheers
